crates/noirc_frontend/src/hir/def_map/mod.rs

```diff
@@ -77,7 +77,24 @@ impl CrateDefMap {
 
         // First parse the root file.
         let root_file_id = context.crate_graph[crate_id].root_file_id;
-        let ast = parse_file(&mut context.file_manager, root_file_id, errors);
+        let mut ast = parse_file(&mut context.file_manager, root_file_id, errors);
+
+        // TODO(#1850): This check should be removed once we fully move over to the new SSA pass
+        // Compiling with the old SSA pass will lead to duplicate method definitions between
+        // the `slice` and `array` modules of the stdlib.
+        //
+        // The last crate represents the stdlib crate.
+        // After resolving the manifest of the local crate the stdlib is added to the manifest and propagated to all crates
+        // thus being the last crate.
```

Comment:
Are you sure the `std` is always the last crate`? It isn't dependent on the dependency order in the toml file at all?

Reply:
This is true as of https://github.com/noir-lang/noir/pull/1840. We shouldn't rely on this behaviour however.

Reply:
Yeah I wasn't a big fan of this check either, but I wasn't sure of the best way to differentiate between two different stdlibs without making two entirely separate stdlibs. If we think this is a safer option though I can just switch to that.